code/modules/antagonists/wizard/grand_ritual/grand_ritual_finale.dm

```diff
@@ -0,0 +1,407 @@
+	for (var/turf/transform_turf as anything in GLOB.station_turfs)
+		if (!chosen_theme.can_convert(transform_turf))
+			continue
+		var/dist = get_dist(start_turf, transform_turf)
+		if (dist > greatest_dist)
+			greatest_dist = dist
+		if (!turfs_to_transform["[dist]"])
+			turfs_to_transform["[dist]"] = list()
+		turfs_to_transform["[dist]"] += transform_turf
+
+	if (chosen_theme.can_convert(start_turf))
+		chosen_theme.apply_theme(start_turf)
+
+	for (var/iterator in 1 to greatest_dist)
+		if(!turfs_to_transform["[iterator]"])
+			continue
+		addtimer(CALLBACK(src, PROC_REF(transform_area), turfs_to_transform["[iterator]"]), (5 SECONDS) * iterator)
+
+/datum/grand_finale/midas/proc/transform_area(list/turfs)
+	for (var/turf/transform_turf as anything in turfs)
+		if (!chosen_theme.can_convert(transform_turf))
+			continue
+		chosen_theme.apply_theme(transform_turf)
+
+/// Kill yourself and probably a bunch of other people
+/datum/grand_finale/armageddon
+	name = "Annihilation"
+	desc = "This crew have offended you beyond the realm of pranks. Make the ultimate sacrifice to teach them a lesson your elders can really respect. \
+		YOU WILL NOT SURVIVE THIS."
+	icon = 'icons/hud/screen_alert.dmi'
+	icon_state = "wounded"
+	minimum_time = 90 MINUTES // This will probably immediately end the round if it gets finished.
+	ritual_invoke_time = 60 SECONDS // Really give the crew some time to interfere with this one.
+	dire_warning = TRUE
+	glow_colour = "#be000048"
+	/// Things to yell before you die
+	var/static/list/possible_last_words = list(
+		"Flames and ruin!",
+		"Dooooooooom!!",
+		"HAHAHAHAHAHA!! AHAHAHAHAHAHAHAHAA!!",
+		"Hee hee hee!! Hoo hoo hoo!! Ha ha haaa!!",
+		"Ohohohohohoho!!",
+		"Cower in fear, puny mortals!",
+		"Tremble before my glory!",
+		"Pick a god and pray!",
+		"It's no use!",
+		"If the gods wanted you to live, they would not have created me!",
+		"God stays in heaven out of fear of what I have created!",
+		"Ruination is come!",
+		"All of creation, bend to my will!",
+	)
+
+/datum/grand_finale/armageddon/trigger(mob/living/carbon/human/invoker)
+	priority_announce(pick(possible_last_words), null, 'sound/magic/voidblink.ogg', sender_override = "[invoker.real_name]")
+	var/turf/current_location = get_turf(invoker)
+	invoker.gib()
+
+	var/static/list/doom_options = list()
+	if (!length(doom_options))
+		doom_options = list(DOOM_SINGULARITY, DOOM_TESLA)
+		if (!SSmapping.config.planetary)
+			doom_options += DOOM_METEORS
+
+	switch(pick(doom_options))
+		if (DOOM_SINGULARITY)
+			var/obj/singularity/singulo = new(current_location)
+			singulo.energy = 300
+		if (DOOM_TESLA)
+			var/obj/energy_ball/tesla = new (current_location)
+			tesla.energy = 200
+		if (DOOM_METEORS)
+			var/datum/dynamic_ruleset/roundstart/meteor/meteors = new()
+			meteors.meteordelay = 0
+			var/datum/game_mode/dynamic/mode = SSticker.mode
+			mode.execute_roundstart_rule(meteors) // Meteors will continue until morale is crushed.
```

Comment:
Surprisingly yeah it does. I did realise when double checking that it doesn't give you any immediate feedback though so I made it print a message so that the player doesn't ahelp thinking that nothing happened.